cumulus_library/upload.py

```diff
@@ -4,24 +4,31 @@
 from pathlib import Path
 
 import requests
+
+from pandas import read_parquet
 from rich.progress import Progress, TaskID
 
 from cumulus_library.helper import get_progress_bar
 
 
 def upload_data(
-    progress: Progress, file_upload_progress: TaskID, file_path: Path, args: dict
+    progress: Progress,
+    file_upload_progress: TaskID,
+    file_path: Path,
+    version: int,
+    args: dict,
 ):
     """Fetches presigned url and uploads file to aggregator"""
     study = file_path.parts[-2]
     file_name = file_path.parts[-1]
     progress.update(file_upload_progress, description=f"Uploading {study}/{file_name}")
-    subscription = file_name.split(".")[0]
+    data_package = file_name.split(".")[0]
     prefetch_res = requests.post(
         args["url"],
         json={
             "study": study,
-            "data_package": subscription,
+            "data_package": data_package,
+            "version": f"{version:03d}",
```

Comment:
Why the `:03d` formatting? That feels like it's for some implicit reason about what the aggregator would want to see? If so, I'd like to see either a comment about it or just send the aggregator an int and it can format it like it wants.

Reply:
So my thought was this:
- I'd like to have sensible alphasorting version numbers, i.e. 001-009, then 010, when viewing data in buckets/transaction metadata
- 3 decimals is a paranoid number about how many changes we should see.
- If you are only looking at the library codebase, it might be more explicit as to what is happening

But - I can see an argument for doing this formatting on the receive side instead of the end side. I could go either way.

Reply:
I'm not done with the other PR, but as far as I can tell, it doesn't care about the format of the string at all - it just places the version on disk (though I suppose it could care about the format in so far as any slashes in it would mess it up real good).

Since the aggregator doesn't care, I think what you've done here makes sense. I just bumped on it because it was an unexpected place for the concern of "the admin of the aggregator would like visually pleasing sorted versions" to appear. (that's a totally valid concern, I just didn't expect it on the library side).

I do this kind of formatting in the ETL too - when writing out ndjson formats, I use 3 decimals. I get the desire.

Your code is fine as-is. You could send an int, which solves some problems (formatting can more properly live on the aggregator side and it doesn't have to check the contents of that string), but the library loses the ability to do minor versions or some other goofy thing. Trade-offs. I might land on the side of "send an int and have aggregator own the how-does-it-look-on-disk logic", but I'm not mad about it.